rust-runtime/aws-smithy-cbor/src/decode.rs

```diff
@@ -75,28 +77,85 @@ impl<'b> Decoder<'b> {
         self.decoder.skip().map_err(DeserializeError::new)
     }
 
-    // TODO An API to support both definite and indefinite strings could be:
-    //
-    //     pub fn str(&mut self, cap: u64) -> Result<&'b str, DeserializeError> {
-    //
-    pub fn str(&mut self) -> Result<&'b str, DeserializeError> {
-        self.decoder.str().map_err(DeserializeError::new)
+    // TODO-David: confirm benchmarks and keep either `str_alt` or `str`.
+    // The following seems to be a bit slower than the one we have kept.
```

Comment:
On my machine `str` is faster than `str_alt` for definite-length, and slower for indefinite-length.

I can see why `str` would be slower for indefinite length strings, since we'd be checking what data type we're decoding twice: once when attempting as a definite length string with `self.decoder.str()`, and another one when delegating to `self.string()`.

I don't understand why `str` would be faster than `str_alt` for _definite_ length strings though.